front50-core/src/main/java/com/netflix/spinnaker/front50/model/plugins/remote/RemoteExtension.java

```diff
@@ -0,0 +1,70 @@
+package com.netflix.spinnaker.front50.model.plugins.remote;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.netflix.spinnaker.front50.model.plugins.PluginInfo;
+import com.netflix.spinnaker.front50.model.plugins.remote.stage.StageRemoteExtensionConfig;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+/**
+ * A Spinnaker plugin's remote extension configuration.
+ *
+ * <p>This model is used by Spinnaker to determine which extension points and services require
+ * remote extension point configuration.
+ */
+@Data
+@NoArgsConstructor
+public class RemoteExtension {
+
+  /**
+   * The remote extension type. The remote extension is configured in the service that implements
+   * this extension type. The plugin release {@link PluginInfo.Release#requires} field is used to
+   * inform Spinnaker which service to use in configuring the remote extension type and if the
+   * remote extension is compatible with the running version of the Spinnaker service.
```

Comment:
Great docs, but I think some of this actually belongs at the top-level class docs, specifically where you start talking about `requires` and you're no longer talking about the `type` property, but how we configure the said `type`. Hope that makes sense?